File: src/main/java/com/example/eyeofsauron/service/RouteService.java
package com.example.eyeofsauron.service;

import com.example.eyeofsauron.entity.Route;
import com.example.eyeofsauron.repository.RouteRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class RouteService {
    private RouteRepository routeRepository;

    public List<Route> getAll() { return routeRepository.findAll(); }

    public Optional<Route> getById(Long id) { return routeRepository.findById(id); }

    public List<Route> getBySecuredFacility(Long id) { return routeRepository.getBySecuredFacilityId(id); }

    public void create(Route route) { routeRepository.save(route); }

    public void update(Route route) { routeRepository.save(route); }

    public void delete(Route route) { routeRepository.delete(route); }

    public void deleteById(Long id) { routeRepository.deleteById(id); }
}

File: src/main/kotlin/com/example/eyeofsauron/service/RouteService.kt
package com.example.eyeofsauron.service

import com.example.eyeofsauron.entity.Route
import com.example.eyeofsauron.repository.RouteRepository
import org.springframework.stereotype.Service

/**
 * Сервис для работы с маршрутами на охраняемых объектах
 * @author rodkinsi
 */
@Service
class RouteService(private val repository: RouteRepository) {
    fun getAll(): List<Route> = repository.findAll()

    fun getById(id: Long) = repository.findById(id)

    fun getBySecuredFacilityId(securedFacilityId: Long) = repository.findBySecuredFacilityId(securedFacilityId)

    fun create(route: Route) = repository.save(route)

    fun update(route: Route) {
        if(!repository.existsById(route.id))
            throw(Exception("Record not found!"))
        repository.save(route)
    }

    fun delete(route: Route) = repository.delete(route)

    fun deleteById(id: Long) = repository.deleteById(id)
}